lib/data.js

```diff
@@ -79,6 +79,57 @@ function init(env, ctx) {
     // if there's no updates done so far, just return the full set
     if (!oldData.sgvs) { return newData; }
 
+    function nsArrayTreatments(oldArray, newArray) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1542425904/issues/source?bid=2452379&fileBranchId=3015078#l82)